### PR TITLE
fix: watch the zk node once the master taskmanager has been changed

### LINKED ISSUE #1066: Update TaskManager Java SDK to initialize with ZK and watch ZK node
Now the Java SDK of TaskManager access BRPC server directly and we should watch th ZK node once the master TaskManager has been changed.

### PR DESCRIPTION
**Pull request by yabg-shuai666**
close #1066

### CODE REVIEW

**Comment by yabg-shuai666** [comment]
It is a problem how to reconnect directly when the content of the master node changes.

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 1]
```diff
@@ -35,8 +44,73 @@
...
     private RpcClient rpcClient;
+    private CuratorFramework zkClient;
     private TaskManagerInterface taskManagerInterface;
+    private static final Log logger = LogFactory.getLog(TaskManagerClient.class);
+
+    public TaskManagerClient(String zkCluster, String zkPath) throws Exception {
+        if (zkCluster == null || zkCluster.length() == 0 || zkPath == null || zkPath.length() == 0) {
+            logger.info("Zookeeper address is wrong, please check the configuration");
+        }
+        String masterZnode = zkPath + "/taskmanager" + "/leader";
```
Comment (aceforeverd) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Merge the two const string

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 2]
```diff
@@ -35,8 +44,73 @@
...
+     * @param curator the zkClient.
+     * @param path the path of zkNode.
+     */
+    public static void watch(CuratorFramework curator, String path) throws Exception {
+        final NodeCache nodeCache = new NodeCache(curator, path, false);
+        nodeCache.getListenable().addListener(new NodeCacheListener() {
+            @Override
+            public void nodeChanged() throws Exception {
+                System.out.println("The content of the node was changed or deleted, please try to reconnect");
+                System.exit(0);
```
Comment (tobegit3hub) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Do not exit when `nodeChanged`. Just reset the rpc client with the new TaskManager server.

**Review by tobegit3hub — CHANGES_REQUESTED** [review]

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 3]
```diff
@@ -34,18 +43,97 @@
...
+        }
+    }
 
+    /**
+     * Create a data change listener event for this node.
+     *
+     * @param curator the zkClient.
+     * @param path the path of zkNode.
+     */
+    public static void watch(CuratorFramework curator, String path) throws Exception {
```
Comment (tobegit3hub) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Remove static for this method.

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 4]
```diff
@@ -34,18 +43,97 @@
  */
 public class TaskManagerClient {
 
-    private RpcClient rpcClient;
-    private TaskManagerInterface taskManagerInterface;
+    private static RpcClient rpcClient;
```
Comment (tobegit3hub) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Make this non-static.

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 5]
```diff
@@ -34,18 +43,97 @@
...
+    private static RpcClientOptions clientOption;
```
Comment (tobegit3hub) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Make this non-static as well.

**Review by tobegit3hub — CHANGES_REQUESTED** [review]

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 6]
```diff
@@ -34,18 +43,97 @@
...
+                    RpcClient rpcClient = TaskManagerClient.rpcClient;
+                    rpcClient.stop();
+
+                    String serviceUrl = "list://" + endpoint;
+                    List<Interceptor> interceptors = new ArrayList<Interceptor>();
+                    rpcClient = new RpcClient(serviceUrl, TaskManagerClient.clientOption, interceptors);
+                    TaskManagerClient.taskManagerInterface = BrpcProxy.getProxy(rpcClient, TaskManagerInterface.class);
+                    RpcContext.getContext().setLogId(1234L);
+                } else {
+                    System.out.println("The content of the node was deleted, please try to reconnect");
```
Comment (aceforeverd) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: log？

Comment (yabg-shuai666) [reply java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Log should be standardized.

**Review by tobegit3hub — APPROVED** [review]
LGTM

File: java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java [hunk 7]
```diff
@@ -35,17 +44,106 @@
...
+
+    /**
+     * Constructor of TaskManager client.
+     *
+     * @param endpoint the endpoint of TaskManager server, for example "127.0.0.1:9902".
+     */
+    public TaskManagerClient(String endpoint) {
 
-        RpcClientOptions clientOption = new RpcClientOptions();
+        clientOption = new RpcClientOptions();
```
Comment (vagetablechicken) [inline java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: why not reuse `rpcConnect()` in `public TaskManagerClient(String endpoint)`?

Comment (yabg-shuai666) [reply java/openmldb-taskmanager/src/main/java/com/_4paradigm/openmldb/taskmanager/client/TaskManagerClient.java]: Ignore the modification just now, it should be changed to the rpcConnect method (forget the rpcConnect method has been changed last week)

**Review by HuilinWu2 — APPROVED** [review]
LGTM

**Pull request merged**